pkg/goDB/DBWorkManager.go

```diff
@@ -108,41 +118,73 @@ func (w *DBWorkManager) CreateWorkerJobs(tfirst int64, tlast int64, query *Query
 
 	// make sure to start with zero workloads as the number of assigned
 	// workloads depends on how many directories have to be read
-	numDirs := 0
-	for i, file := range dirList {
+	var (
+		numDirs int
+		curDir  *gpfile.GPDir
+	)
+	workloadBulk := make([]*gpfile.GPDir, 0, WorkBulkSize)
+	for _, file := range dirList {
 		if file.IsDir() && (file.Name() != "./" || file.Name() != "../") {
 			dirName = file.Name()
-			tempdirTstamp, _ := strconv.ParseInt(dirName, 10, 64)
+			tempdirTstamp, err := strconv.ParseInt(dirName, 10, 64)
+			if err != nil {
+				return false, fmt.Errorf("failed to parse timestamp from directory `%s`: %w", dirName, err)
```

Comment:
What would that do? There's usually no escaped quotes in `dirName`, I just want to keep it readable in case it's an empty string. Or am I missing something here?